design-documents/frontend/technical-vision.md

```diff
@@ -0,0 +1,140 @@
+
+### Business Logic Repository
+1. Written and [documented](https://github.com/Microsoft/tsdoc#tsdoc) in [TypeScript][].
+1. [Semantic versioning](https://semver.org/) to ensure that progress can be made without breaking things for existing - consumers.
+1. 100% unit test coverage via [Jest][] for **consistency**.
+
+For **performance** reasons, JavaScript should only be used for cheap operations on the front end, reserving expensive operations for a services layer. It's worth calling out that we could achieve even more code sharing if this API were written in JavaScript. For practical reasons, however, it makes more sense to leverage internal PHP resources for this task.
+
+## Browser Support
+All users should have a **consistent** user experience. For the best **performance**, those with the latest browsers should not be penalized by incurring an additional payload hit than those with older browsers. To achieve this goal, we download just the [polyfills](https://remysharp.com/2010/10/08/what-is-a-polyfill) needed for our sites, tailored to each browser via [Polyfill.io](https://polyfill.io) feature detection.
+
+> We test browser support with a test suite for each feature. A browser is considered compliant only if it passes all the tests, so this is a reasonably comprehensive test rather than a naïve feature-detect.
+
+_TODO: Investigate building two separate [TypeScript][] builds, one that compiles down to ES5 and another lighter-weight modern build, specifically for modern browsers. Depending on how well Polyfill.io works, the modern build might be all we need._
+
+## React UI Component Library
+
+_Consistency, Productivity, Scalability, Maintainability, Accessibility._
+
+We should lean against downloading various UI components from different sources for the following reasons:
+
+- **Inconsistent** implementations, documentation, theming and customization.
+- Limited **accessibility** and support.
+
+[KendoReact | UI for React Developers](https://www.telerik.com/kendo-react-ui/) is a great option for those wishing to get React UI components from the same source. Still, there will be times when you need to download a 3rd-party package.
+
+### Downloading a 3rd-party package
+
+You may have to download a 3rd party component. When this happens, consider the following criteria before making a final choice:
+
+1. First thing's first – what is the license?
+   - Is it open source?
+   - You may need to reach out to your legal department if things are a bit blurry.
+1. Does the package have a bright future?
+   - Popularity – are there a significant number of weekly downloads?
+   - Is the downloads line graph going up or down?
+1. Look for tests, ideally with 100% test coverage.
+1. Look at install size with [Package Phobia](https://packagephobia.now.sh/).

```

Comment:
Probably meant Bundle Phobia here

Reply:
Addressed in 4f625a47b50e0c0ac67ccbf5fed03eff17b94d40